### PR TITLE
SCI: Move MidiDriver_CMS to the new CMS API

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

**Comment by athrxx** [comment]
Okay, I tested Colonel's Bequest with this PR and noticed that the sound is a bit off.

The (hopefully only) reason seems to be that you killed the proper envelope timing in lines 921-924. When I restore that to what it was it seems to fix the issue.

Haven't tested anything else yet. Which games did you test yourself?

**Comment by ccawley2011** [comment]
> The (hopefully only) reason seems to be that you killed the proper envelope timing in lines 921-924. When I restore that to what it was it seems to fix the issue.

Would specifying a frequency of 60 in `_cms->start()` instead of the default value of 250 accomplish the same effect?

**Comment by athrxx** [comment]
> Would specifying a frequency of 60 in `_cms->start()` instead of the default value of 250 accomplish the same effect?

I just did a quick test. Unfortuantely, it will slow down the midi parsing (the call to `(*_timerProc)(_timerParam);`), too. So, if you want to go that way, you'll have to adjust that...

**Comment by ccawley2011** [comment]
OK, I've restored the removed code.

**Review by athrxx — APPROVED** [review]
Now it seems to be fine.

**Comment by NMIError** [comment]
Good change! I've tested this with Space Quest 3; sounds the same as the original.

File: engines/sci/sound/drivers/cms.cpp [hunk 1]
```diff
@@ -272,8 +272,9 @@ void CMSVoice::sendFrequency() {
 }
 
 void CMSVoice::cmsWrite(uint8 reg, uint8 val) {
-	_cms->portWrite(0x221 + _portOffset, reg);
-	_cms->portWrite(0x220 + _portOffset, val);
+	/* TODO: Call _cms->writeReg instead? */
+	_cms->write(0x221 + _portOffset, reg);
+	_cms->write(0x220 + _portOffset, val);
```
Comment (NMIError) [inline engines/sci/sound/drivers/cms.cpp:277]: Are you going to change this to writeReg as part of this PR? ScummVM does not need to use I/O ports of course; would be nice to get rid of them.

Comment (ccawley2011) [reply engines/sci/sound/drivers/cms.cpp]: OK, done.

**Review by NMIError — APPROVED** [review]
Great, thanks!

**Pull request merged**